### PR TITLE
feat: implement save api key as editor mechanism

### PR DESCRIPTION
**Pull request by hvn2k1**
Part of https://github.com/bucketeer-io/bucketeer/issues/1331

- Save API key in context and extract it to embed into editor
- Add user name in access token to embed into editor

### CODE REVIEW

File: pkg/account/domain/account.go [hunk 1]
```diff
@@ -305,6 +305,19 @@ func (a *AccountV2) resetDefaultFilter(targetFilter proto.FilterTargetType, envi
 	}
 }
 
+func GetAccountFullName(a *proto.AccountV2) string {
+	if a == nil {
+		return ""
+	}
```
Comment (cre8ivejp) [inline pkg/account/domain/account.go:311]:
```suggestion
func (a *AccountV2) GetAccountFullName() string {
```

File: pkg/auth/api/api.go [hunk 2]
```diff
@@ -458,6 +459,7 @@ func (s *authService) generateToken(
 		Expiry:        accessTokenTTL,
 		IssuedAt:      timeNow,
 		Email:         userEmail,
+		Name:          domain.GetAccountFullName(account.Account),
```
Comment (cre8ivejp) [inline pkg/auth/api/api.go:462]:
```suggestion
		Name:          account.GetAccountFullName(),
```

File: pkg/role/role.go [hunk 3]
```diff
@@ -55,8 +63,22 @@ func CheckEnvironmentRole(
 	if !ok {
 		return nil, ErrUnauthenticated
 	}
+	publicAPIEditor := getAPIKeyEditor(ctx)
+	if publicAPIEditor != nil && publicAPIEditor.Token != "" && token.IsSystemAdmin {
+		var accountName string
+		account, err := getAccountFunc(publicAPIEditor.Maintainer)
+		if err == nil && account != nil {
+			accountName = accdomain.GetAccountFullName(account)
```
Comment (cre8ivejp) [inline pkg/role/role.go:71]:
```suggestion
			accountName = account.GetAccountFullName()
```

File: pkg/role/role.go [hunk 4]
```diff
@@ -105,9 +129,24 @@ func CheckOrganizationRole(
...
+		var accountName string
+		resp, err := getAccountFunc(publicAPIEditor.Maintainer)
+		if err == nil && resp != nil && resp.Account != nil {
+			accountName = accdomain.GetAccountFullName(resp.Account)
+		}
+		return &eventproto.Editor{
+			Email:           publicAPIEditor.Maintainer,
+			Name:            accountName,
+			PublicApiEditor: publicAPIEditor,
+		}, nil
```
Comment (cre8ivejp) [inline pkg/role/role.go:143]:
```suggestion
		account, err := getAccountFunc(publicAPIEditor.Maintainer)
		if err != nil {
		        return err // return an internal error
		}
		return &eventproto.Editor{
			Email:           publicAPIEditor.Maintainer,
			Name:            account.GetAccountFullName(),
			PublicApiEditor: publicAPIEditor,
		}, nil
```

File: pkg/account/domain/account.go [hunk 5]
```diff
@@ -305,6 +305,19 @@ func (a *AccountV2) resetDefaultFilter(targetFilter proto.FilterTargetType, envi
 	}
 }
 
+func (a *AccountV2) GetAccountFullName() string {
+	if a == nil {
+		return ""
+	}
```
Comment (cre8ivejp) [inline pkg/account/domain/account.go:311]: It can't be nil here.

```suggestion
```

Comment (hvn2k1) [reply pkg/account/domain/account.go]: I have fixed that and added test cases for the method

**Review by cre8ivejp — APPROVED** [review]
Great work 🎉

**Pull request merged**